src/ariel-os-alloc/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+//! This crate glues `embedded-alloc` into Ariel OS.
+
+#![no_std]
+#![deny(missing_docs)]
+#![deny(clippy::pedantic)]
+// required for tests:
+#![cfg_attr(test, no_main)]
+#![cfg_attr(test, feature(impl_trait_in_assoc_type))]
+#![cfg_attr(test, feature(used_with_arg))]
+
+// With embedded-test enabled, this crate gets built *twice*, once regularly
+// as the system alloc it is supposed to be, and once as test application.
+// In the latter case, `cfg(test)` is set.
+// So we *only* set up the global stuff if *not* testing in order to avoid clashes.
+#[cfg(not(test))]
+pub use alloc::init;
+
+#[cfg(not(test))]
+mod alloc {
+    /// Initialize the heap.
+    ///
+    /// This is called by `ariel-os-rt` early during system initialization.
+    ///
+    /// # Safety
+    ///
+    /// Call only once!
+    #[cfg(context = "cortex-m")]
```

Comment:
Differences between the platforms (at least: "the comment is the same") could be easier to grasp if there was a single `pub unsafe init() {}` function that called a platform specific specific function.

When things go wrong, this makes it easier to see that this is an internal problem, for the pub surface of the module is not arch dependent.

Reply:
I created `init_embedded_alloc()` and `init_esp_alloc()`.